### ISSUE TITLE
hickle.dump raises FileError when passed a file opened in binary mode

### ISSUE THREAD

**Issue by jbweston** [telegraphic/hickle#123]
## Expected behaviour
`hickle.dump` suceeds regardless of whether the file was opened in mode `w` or `wb`

## Steps to reproduce
```
import hickle

# Works fine
with open("whatever", "w") as f:
    hickle.dump(None, f)

# Raises FileError
with open("whatever", "wb") as f:
    hickle.dump(None, f)
```

## Platform information

Python:  3.7.4
hickle: 3.4.5

**Comment by jbweston** [comment]
This fails because there are explicit `isinstance` checks [here](https://github.com/telegraphic/hickle/blob/1419babeb57c84d0b6ff4a8f89b193defe45629e/hickle/hickle.py#L177).  These should either be extended to add `io.BufferedWriter`, replaced with duck typing.

**Comment by jbweston** [comment]
Also if/when we bump the required `h5py` version to `2.9` we can pass file objects directly to `h5py.File` ([see here](http://docs.h5py.org/en/stable/high/file.html#python-file-like-objects)).

**Comment by telegraphic** [comment]
If it's just as simple to check for `io.BufferedWriter`, happy to change.

For completeness, I note that the file I/O in hickle is done with `h5py`, which doesn't have a `wb` flag:
http://docs.h5py.org/en/stable/high/file.html#opening-creating-files

Can you see any reason why an end user would mind that `h5py.File` was being called with `w` instead of the `wb` of the file object passed?  (Seems reasonable to me!)

**Comment by jbweston** [comment]
> If it's just as simple to check for io.BufferedWriter, happy to change.

Thanks!

It makes sense that the `b` part of the mode be ignored because it does not mean anything for HDF5 files. If anyone minds then they would need to specify what exactly they are expecting to happen.

The reason why I want to be able to pass an open `wb`-file to hickle is because I dispatch to different serializers (e.g. pickle, dill...) and I did not want to special-case my code for hickle. It makes sense to me, given that the file handle is not used anyway, that if you can pass a file opened in text mode then you should be able to pass a file in binary mode.

**Comment by jbweston** [comment]
Thanks! I consider this solved in light of 50b16fc
